src/App.tsx

```diff
@@ -1,26 +1,38 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import { Route, Switch, useHistory } from 'react-router-dom'
+import { useDispatch, useSelector } from 'react-redux'
+import { Home, SignIn, TypingTest, Scorecard } from './pages'
+import { AppBar } from './components'
+import {
+  HOME_ROUTE,
+  SIGN_IN_ROUTE,
+  TYPING_TEST_ROUTE,
+  SCORECARD_ROUTE,
+} from './constants'
+import { userLogout } from './redux'
+import { PrivateRoute } from './routes'
+import './App.css'
 
 function App() {
+  const dispatch = useDispatch()
+  const userLogin = useSelector((state: any) => state.userLogin)
```

Comment:
Typing for store state is missing? We should create and use types for store https://react-redux.js.org/using-react-redux/usage-with-typescript#typing-hooks-manually.